x/auth/vesting/types/vesting_account_test.go

```diff
@@ -868,134 +862,100 @@ func TestClawback(t *testing.T) {
-	_, err = app.StakingKeeper.Delegate(ctx, addr, sdk.NewInt(65), stakingtypes.Unbonded, val, true)
-	require.Error(t, err)
-
-	// undelegation should emit an error(delegator does not contain delegation)
-	_, err = app.StakingKeeper.Undelegate(ctx, addr, valAddr, sdk.NewDec(5))
-	require.Error(t, err)
-
-	// at 11hr, its all still locked, but 600fee, 50stake can be clawed back
-	ctx = ctx.WithBlockTime(now.Add(11 * time.Hour))
-	_, _, dest := testdata.KeyTestPubAddr()
-	va2 := app.AccountKeeper.GetAccount(ctx, addr).(*types.ClawbackVestingAccount)
-	clawbackAction := types.NewClawbackAction(funder, dest, app.AccountKeeper, app.BankKeeper)
-	err = va2.Clawback(ctx, clawbackAction)
-	require.NoError(t, err)
-
-	// check vesting account, 400fee, 50stake should remain
-	// dest should get 600fee, 50stake
-	addrBalances := app.BankKeeper.GetAllBalances(ctx, addr)
-	expectedAddrCoins := sdk.Coins{sdk.NewInt64Coin(feeDenom, 400), sdk.NewInt64Coin(stakeDenom, 50)}
-	require.Equal(t, expectedAddrCoins, addrBalances)
-	destBalances := app.BankKeeper.GetAllBalances(ctx, dest)
-	expectedDestCoins := sdk.Coins{sdk.NewInt64Coin(feeDenom, 600), sdk.NewInt64Coin(stakeDenom, 50)}
-	require.Equal(t, expectedDestCoins, destBalances)
-	// none of the remaining addr coins should be spendable
-	require.True(t, app.BankKeeper.SpendableCoins(ctx, addr).Empty())
-
-	// at hour 12, the unlock should finish and the account should have its 400fee, 50 stake liquid
-	ctx = ctx.WithBlockTime(now.Add(15 * time.Hour))
-	va2 = app.AccountKeeper.GetAccount(ctx, addr).(*types.ClawbackVestingAccount)
-	require.Equal(t, expectedAddrCoins, app.BankKeeper.SpendableCoins(ctx, addr))
-}
-
-func TestCliffClawback(t *testing.T) {
-	c := sdk.NewCoins
-	fee := func(x int64) sdk.Coin { return sdk.NewInt64Coin(feeDenom, x) }
-	stake := func(x int64) sdk.Coin { return sdk.NewInt64Coin(stakeDenom, x) }
-	now := tmtime.Now()
-
-	// set up simapp and validators
-	app := simapp.Setup(false)
-	ctx := app.BaseApp.NewContext(false, tmproto.Header{}).WithBlockTime((now))
-	valAddr, val := createValidator(t, ctx, app, 100)
-	require.Equal(t, "stake", app.StakingKeeper.BondDenom(ctx))
-
-	totalVestingCoins := sdk.NewCoins(sdk.NewInt64Coin(feeDenom, 1000), sdk.NewInt64Coin(stakeDenom, 100))
-	lockupPeriods := []types.Period{
-		{Length: int64(12 * 3600), Amount: c(fee(1000), stake(100))},
-	}
-	vestingPeriods := []types.Period{
-		{Length: int64(8 * 3600), Amount: c(fee(200))},            // 8am
-		{Length: int64(1 * 3600), Amount: c(fee(200), stake(50))}, // 9am
-		{Length: int64(6 * 3600), Amount: c(fee(200), stake(50))}, // 3pm
-		{Length: int64(2 * 3600), Amount: c(fee(200))},            // 5pm
-		{Length: int64(1 * 3600), Amount: c(fee(200))},            // 6pm
+	testCases := []struct {
+		name              string
+		ctxTime           time.Time
+		vestingAccBalance sdk.Coins
+		spendableCoins    sdk.Coins
+		funderBalance     sdk.Coins
+	}{
+		// fails
```

Comment:
Actually, is there not a way we can use expected errors or something rather than simply commenting above `fails` each test case?

E.g. a method to call with some expected result that would indicate "failure".

Reply:
My original intention was to mark that this is failing even though it's supposed to pass.
I was worried that using something like `expected errors`(a boolean) and setting it to false would confuse us that it's supposed to fail even though it's not.

Reply:
Gotcha. I think relying solely on that comment is not sufficient or all too helpful. Perhaps we can still use a boolean, but just name it something different?

Reply:
My comment / code patch suggestion should fix this, the test case was expecting behavior that isn't right!